src/components/ProtectedOwnerRoute.jsx

```diff
@@ -0,0 +1,7 @@
+import { Navigate } from 'react-router-dom';
+import { useAuth } from './AuthProvider';
+
+export default function ProtectedOwnerRoute({ children }) {
+    const { role } = useAuth();
+    return role === 'owner' ? children : <Navigate to = "/" replace />;
```

Comment:
Makes ProtectedOwnerRoute where it specifically checks if the role of the current user is owner.